src/LinearSolvers/preconditioners.jl

```diff
@@ -55,18 +79,28 @@ mutable struct BlockJacobiPreconditioner <: AbstractPreconditioner
         for (i,v) in enumerate(part)
             push!(partitions[v], i)
         end
-        Js = Vector{Matrix{Float64}}(undef, npart)
-        nJs = maximum(length.(partitions))
-        id = Matrix{Float64}(I, nJs, nJs)
+        lpartitions = VI(undef, npart)
+        for i in 1:npart
+            lpartitions[i] = length(partitions[i])
+        end
+        blocksize = maximum(length.(partitions))
```

Comment:
In fact, you did the broadcasting here :-) I would rephrase this line as 
`blocksize = maximum(lpartitions)`
to avoid doing twice the same operation

Reply:
lol psst.